perses/app/relative_setup.py

```diff
@@ -159,10 +160,46 @@ def __init__(self,
         self._solvent_box_dimensions = solvent_box_dimensions
         self._use_given_geometries = use_given_geometries
         self._given_geometries_tolerance = given_geometries_tolerance
-        self._ligand_input = ligand_input
+        self.small_molecule_forcefield = small_molecule_forcefield
+        if ligand_input:
+            self._ligand_input = AnyPath(ligand_input)
+
+        if protein_pdb_filename:
+            self.protein_pdb_filename = AnyPath(protein_pdb_filename)
+        else:
+            self.protein_pdb_filename = protein_pdb_filename
+
+        if receptor_mol2_filename:
+            self.receptor_mol2_filename = AnyPath(receptor_mol2_filename)
+        else:
+            self.receptor_mol2_filename = receptor_mol2_filename
```

Comment:
As far as I can see this effectively disables reading protein structures from mol2, since always having the `.protein_pdb_filename` attributed will avoid entering the `mol2` branch [here](https://github.com/choderalab/perses/blob/2e3a7a54d0adbc7d751574f66fe0b48799eb7bd3/perses/app/relative_setup.py#L592-L609), we probably need a better way of handling the protein input (check the TODO and comment in the same lines).

Reply:
Ah I can see why, this doesn't change the behavior in the end, but we do need a better way to deal with reading from different formats as per the TODO comment.